### PR TITLE
[Tests-Only] add endpoints tests

### PR DESCRIPTION
**Pull request by kiranparajuli589**
With this PR:
- added tests for endpoint status and behavior
- run the tests on CI
Closes https://github.com/owncloud/owncloud-test-middleware/issues/38

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+							"POSTGRES_PASSWORD": getDbPassword(db),
+							"POSTGRES_DB": getDbDatabase(db),
+					},
+			}]
+
+	if dbName == "oracle":
+			return [{
+					"name": dbName,
+					"image": "deepdiver/docker-oracle-xe-11g:latest",
+					"pull": "always",
```
Comment (dpakach) [inline .drone.star:211]: it is not required here IMO. just use MySQL database by default

File: .drone.star [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+
+
+def getDbDatabase(db):
+	name = getDbName(db)
+
+	if name == "oracle":
+			return "XE"
+
+	return "owncloud"
+
```
Comment (dpakach) [inline .drone.star:178]: we dont need to test with different databases here, please use MySQL only

File: .drone.star [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+			return [{
+					"name": dbName,
+					"image": db,
+					"pull": "always",
+					"environment": {
+							"MYSQL_USER": getDbUsername(db),
+							"MYSQL_PASSWORD": getDbPassword(db),
+							"MYSQL_DATABASE": getDbDatabase(db),
+							"MYSQL_ROOT_PASSWORD": getDbRootPassword(),
+					},
```
Comment (dpakach) [inline .drone.star:192]: the indentation is messed up everywhere, please fix this whole file.

File: package.json [hunk 4]
```diff
@@ -7,11 +7,13 @@
...
   "scripts": {
-    "start": "node src/app.js",
-    "watch": "nodemon src/app.js",
+    "start": "node src/index.js",
+    "watch": "nodemon src/index.js",
     "lint": "eslint src --ext js --color",
     "lint-fix": "eslint src --ext js --color --fix",
-    "test": "jest */**/*.test.js"
+    "test": "jest */**/*.test.js --testPathIgnorePatterns=tests/",
+    "integration-tests:endpoints": "jest tests/endpoint",
```
Comment (dpakach) [inline package.json:15]: change it to `test:integration`

File: package.json [hunk 5]
```diff
@@ -7,11 +7,13 @@
...
   "license": "ISC",
   "scripts": {
-    "start": "node src/app.js",
-    "watch": "nodemon src/app.js",
+    "start": "node src/index.js",
+    "watch": "nodemon src/index.js",
     "lint": "eslint src --ext js --color",
     "lint-fix": "eslint src --ext js --color --fix",
-    "test": "jest */**/*.test.js"
+    "test": "jest */**/*.test.js --testPathIgnorePatterns=tests/",
```
Comment (dpakach) [inline package.json:14]:
```suggestion
    "test:unit": "jest */**/*.test.js --testPathIgnorePatterns=tests/",
```

File: tests/testingApp/testingAppRequired.test.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+describe('init should not be accessible without testing app enabled on the server', () => {
+  // beforeEach(async () => {
+  //   await runOccWithCli(['app:disable', 'testing'])
+  // })
+
+  // afterEach(async () => {
+  //   // middle did not initialized successfully so no need of cleanup
+  //   await runOccWithCli(['app:enable', 'testing'])
+  // })
```
Comment (dpakach) [inline tests/testingApp/testingAppRequired.test.js:14]: remove these

**Review by dpakach — CHANGES_REQUESTED** [review]

**Review by dpakach — COMMENTED** [review]
lgtm 👍🏼

File: .drone.yml [hunk 7]
```diff
@@ -3,33 +3,263 @@ kind: pipeline
 type: docker
```
Comment (individual-it) [inline .drone.yml:3]: why do we have a .yml and .star ?

**Review by individual-it — CHANGES_REQUESTED** [review]

File: .drone.yml (hunk 7 above)
Comment (kiranparajuli589) [reply .drone.yml:3]: the repo seems to use the `.yml` file for ci, it ignores if only the `.star` is pushed. I've used the `.star` file for generating purposes only.

Comment (individual-it) [reply .drone.yml:3]: that is easy to change, I've done that. the .star file should be used now

**Review by individual-it — CHANGES_REQUESTED** [review]
drone.yml file should not be needed anymore

**Review by individual-it — APPROVED** [review]

File: .drone.star [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+
+
+def setupServer(testingServer, tesingAppRequired=True):
+	return [{
+		"name": "setup-server",
+		"image": "owncloudci/php:7.4",
+		"pull": "always",
+		"commands": [
+			"cd /var/www/owncloud/server",
+			"php occ a:e testing" if tesingAppRequired else "php occ a:l testing",
```
Comment (phil-davis) [inline .drone.star:234]:
```suggestion
			"php occ a:e testing" if testingAppRequired else "php occ a:l testing",
```

**Review by phil-davis — COMMENTED** [review]
Just a small typo that does not change anything.

**Review by phil-davis — APPROVED** [review]

**Pull request merged**